paper-server/patches/sources/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java.patch

```diff
@@ -0,0 +1,33 @@
+--- a/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
++++ b/net/minecraft/world/entity/ai/behavior/PoiCompetitorScan.java
+@@ -22,13 +_,23 @@
+                         level.getPoiManager()
+                             .getType(globalPos.pos())
+                             .ifPresent(
+-                                poi -> instance.<List<LivingEntity>>get(nearestLivingEntities)
+-                                    .stream()
+-                                    .filter(entity -> entity instanceof Villager && entity != villager)
+-                                    .map(entity -> (Villager)entity)
+-                                    .filter(LivingEntity::isAlive)
+-                                    .filter(v -> competesForSameJobsite(globalPos, poi, v))
+-                                    .reduce(villager, PoiCompetitorScan::selectWinner)
++                                poi -> {
++                                    List<LivingEntity> livingEntities = instance.get(nearestLivingEntities);
++
++                                    Villager winner = villager;
++                                    for (LivingEntity entity : livingEntities) {
++                                        if (!(entity instanceof Villager && entity != villager)) {
```

Comment:
This can use a pattern instanceof, plus the entity != villager can be taken out of the negated part to be easier to read as was shortly talked about in discord

Reply:
You are right, I forgot about that, will do.